src/app/shared/subscriptions/subscription-modal/subscription-modal.component.ts

```diff
@@ -0,0 +1,260 @@
+      shareReplay(),
+    ).subscribe((ePersonId: string) => {
+      this.ePersonId = ePersonId;
+      if (isNotEmpty(this.subscription)) {
+        this.initFormByGivenSubscription();
+      } else {
+        this.initFormByAllSubscriptions();
+      }
+    });
+  }
+
+  private initFormByAllSubscriptions(): void {
+    this.subscriptionForm = new FormGroup({});
+    for (let t of this.subscriptionDefaultTypes) {
+      const formGroup = new FormGroup({});
+      formGroup.addControl('subscriptionId', this.formBuilder.control(''));
+      formGroup.addControl('frequencies', this.formBuilder.group({}));
+      for (let f of this.frequencyDefaultValues) {
+        (formGroup.controls.frequencies as FormGroup).addControl(f, this.formBuilder.control(false));
+      }
+      this.subscriptionForm.addControl(t, formGroup);
+    }
+
+    this.initFormDataBySubscriptions();
+  }
+
+  /**
+   * If the subscription is passed start the form with the information of subscription
+   */
+  initFormByGivenSubscription(): void {
+    const formGroup = new FormGroup({});
+    formGroup.addControl('subscriptionId', this.formBuilder.control(this.subscription.id));
+    formGroup.addControl('frequencies', this.formBuilder.group({}));
+    (formGroup.get('frequencies') as FormGroup).addValidators(Validators.required);
+    for (let f of this.frequencyDefaultValues) {
+      const value = findIndex(this.subscription.subscriptionParameterList, ['value', f]) !== -1;
+      (formGroup.controls.frequencies as FormGroup).addControl(f, this.formBuilder.control(value));
+    }
+
+    this.subscriptionForm = this.formBuilder.group({
+      [this.subscription.subscriptionType]: formGroup
+    });
+  }
+
+  /**
+   * Get subscription for the eperson & dso object relation
+   * If no subscription start with an empty form
+   */
+  initFormDataBySubscriptions(): void {
+    this.processing$.next(true);
+    this.subscriptionService.getSubscriptionsByPersonDSO(this.ePersonId, this.dso?.uuid).pipe(
+      getFirstSucceededRemoteDataPayload(),
+    ).subscribe({
+      next: (res: PaginatedList<Subscription>) => {
+        if (res.pageInfo.totalElements > 0) {
+          for (let subscription of res.page) {
+            const type = subscription.subscriptionType;
+            const subscriptionGroup: FormGroup = this.subscriptionForm.get(type) as FormGroup;
+            if (isNotEmpty(subscriptionGroup)) {
+              subscriptionGroup.controls.subscriptionId.setValue(subscription.id);
+              for (let parameter of subscription.subscriptionParameterList.filter((p) => p.name === 'frequency')) {
+                (subscriptionGroup.controls.frequencies as FormGroup).controls[parameter.value]?.setValue(true);
+              }
+            }
+          }
+        }
+        this.processing$.next(false);
+      },
+      error: err => {
+        this.processing$.next(false);
+      }
+    });
+  }
+
+  /**
+   * Create/update subscriptions if needed
+   */
+  submit() {
```

Comment:
This entire `submit()` method has a lot of messages from Codecov about missing tests/specs.  It looks like we are missing tests for what should happen when the user clicks "submit".  Right now, the specs only look to test that the form loads properly in the modal window.